nimble/host/src/ble_gap.c

```diff
@@ -998,7 +998,7 @@ ble_gap_update_next_exp(int32_t *out_ticks_from_now)
 
 }
 
-#if MYNEWT_VAL(BLE_ROLE_CENTRAL)
+#if MYNEWT_VAL(BLE_ROLE_CENTRAL) || MYNEWT_VAL(BLE_ROLE_OBSERVER)
```

Comment:
maybe we just need here `#if NIMBLE_BLE_SCAN` ?

Reply:
Yep, that's cleaner. Updated the PR.